crates/router/src/configs/settings.rs

```diff
@@ -393,6 +395,24 @@ pub struct NetworkTransactionIdSupportedConnectors {
     pub connector_list: HashSet<enums::Connector>,
 }
 
+#[derive(Debug, Deserialize, Clone, Default)]
+pub struct NetworkTokenizationSupportedCardNetworks {
+    #[serde(deserialize_with = "deserialize_hashset")]
+    pub card_networks: HashSet<enums::CardNetwork>,
+}
+
+#[derive(Debug, Deserialize, Clone, Default)]
+pub struct NetworkTokenizationService {
+    pub generate_token_url: String,
+    pub fetch_token_url: String,
+    pub token_service_api_key: Secret<String>,
+    pub public_key: Secret<String>,
+    pub private_key: Secret<String>,
+    pub key_id: String,
+    pub delete_token_url: String,
+    pub check_token_status_url: String,
```

Comment:
If we do end up making the `network_tokenization_service` field an `Option` in the `Settings` struct, should we consider validating that these URLs, key ID, keys, etc. are not empty? That way, we would be sure that either the entire section is not specified in config, or valid values are provided for all these fields.